community/consistency-check/src/main/java/org/neo4j/consistency/checking/RelationshipRecordCheck.java

```diff
@@ -19,32 +19,77 @@
 import org.neo4j.kernel.impl.store.record.RelationshipTypeTokenRecord;
 
-class RelationshipRecordCheck
-        extends PrimitiveRecordCheck<RelationshipRecord, ConsistencyReport.RelationshipConsistencyReport>
+public class RelationshipRecordCheck extends
+        PrimitiveRecordCheck<RelationshipRecord,ConsistencyReport.RelationshipConsistencyReport>
 {
-    RelationshipRecordCheck()
+    public static final int CACHE_SLOT_SOURCE_OR_TARGET = 0;
+    public static final int CACHE_SLOT_PREV_OR_NEXT = 1;
+    public static final int CACHE_SLOT_RELATIONSHIP_ID = 2;
+    public static final int CACHE_SLOT_REFERENCE = 3;
+    public static final long SOURCE = 0;
+    public static final long TARGET = -1;
+    public static final long PREV = 0;
+    public static final long NEXT = -1;
+
+    public RelationshipRecordCheck()
     {
-        super( RelationshipTypeField.RELATIONSHIP_TYPE,
-               NodeField.SOURCE, RelationshipField.SOURCE_PREV, RelationshipField.SOURCE_NEXT,
-               NodeField.TARGET, RelationshipField.TARGET_PREV, RelationshipField.TARGET_NEXT );
+        super( true, RelationshipTypeField.RELATIONSHIP_TYPE, NodeField.SOURCE, RelationshipField.SOURCE_PREV,
+                RelationshipField.SOURCE_NEXT, NodeField.TARGET, RelationshipField.TARGET_PREV,
+                RelationshipField.TARGET_NEXT );
     }
 
-    private enum RelationshipTypeField implements
-            RecordField<RelationshipRecord, ConsistencyReport.RelationshipConsistencyReport>,
-            ComparativeRecordChecker<RelationshipRecord, RelationshipTypeTokenRecord, ConsistencyReport.RelationshipConsistencyReport>
+    @SafeVarargs
+    private RelationshipRecordCheck( boolean firstProperty,
+            RecordField<RelationshipRecord,ConsistencyReport.RelationshipConsistencyReport>... fields )
     {
-        RELATIONSHIP_TYPE;
+        super( firstProperty, fields );
+    }
+
+    public static RelationshipRecordCheck RelationshipRecordCheckPass1( boolean firstProperty )
+    {
+        return new RelationshipRecordCheck( firstProperty, RelationshipTypeField.RELATIONSHIP_TYPE );
+    }
 
+    public static RelationshipRecordCheck RelationshipRecordCheckPass2( boolean firstProperty )
+    {
+        return new RelationshipRecordCheck( firstProperty, NodeField.SOURCE, NodeField.TARGET );
+    }
+
+    public static RelationshipRecordCheck RelationshipRecordCheckSourceChain( boolean firstProperty )
+    {
+        return new RelationshipRecordCheck( firstProperty, RelationshipField.SOURCE_NEXT,
+                RelationshipField.SOURCE_PREV, RelationshipField.TARGET_NEXT, RelationshipField.TARGET_PREV,
+                RelationshipField.CACHE_VALUES );
+    }
+
+    public static RelationshipRecordCheck RelationshipRecordCheckTargetChain( boolean firstProperty )
+    {
+        return new RelationshipRecordCheck( firstProperty, RelationshipField.TARGET_NEXT, RelationshipField.TARGET_PREV );
+    }
+
+    private enum RelationshipTypeField
```

Comment:
It seems to be a common pattern here that *RecordCheck classes have their enum fields as internal classes... all except NodeField which is an enum for RelationshipRecordCheck. It's not strictly necessary, is it? We unit test on RecordCheck level anyway.